repoman/git/repository.py

```diff
@@ -120,6 +120,41 @@ def commit(self):
         return self.repository.tip()
 
 
+class GitMergeSquash(GitMerge):
+    def __init__(self, *args, **kwargs):
+        super(GitMergeSquash, self).__init__(*args, **kwargs)
+
+    def perform(self):
+        self._validate_local_branch()
+
+        self._git('merge', '--squash',
+                  self.other_rev.hash,
+                  _ok_code=[0, 1])
+
+        conflicts = self._git('diff', name_only=True, diff_filter='U').split()
+
+        if conflicts:
+            raise MergeConflictError("Conflicts found: merging %s failed" %
+                                     ", ".join(conflicts))
+
+
+class GitMergeRebase(GitMerge):
+    def __init__(self, *args, **kwargs):
+        super(GitMergeSquash, self).__init__(*args, **kwargs)
+
+    def perform(self):
+        self._validate_local_branch()
+
+        self._git('merge', '--rebase',
+                  self.other_rev.hash,
+                  _ok_code=[0, 1])
+
```

Comment:
I don't see this parameter available in `git merge` command man page:
https://git-scm.com/docs/git-merge

I guess it should be done in two steps:
1. do the rebase of the given branch onto the default branch (master, main,...)
2. merge the rebased branch onto the default branch

Morever, it can be to enforced this behaviour by just allowing fast forwarded merges with the parameter `--ff-only`:
```bash
git merge --ff-only
```

Reply:
I think we can do it by running a "git rebase <other branch>", that is the result i was looking for